src/core/annotation.js

```diff
@@ -1359,26 +1369,41 @@ class WidgetAnnotation extends Annotation {
     return [initialState.font, initialState.fontName, initialState.fontSize];
   }
 
-  _computeFontSize(font, fontName, fontSize, height) {
+  _computeFontSize(font, fontName, fontSize, height, lineCount) {
     if (fontSize === null || fontSize === 0) {
-      const em = font.charsToGlyphs("M")[0].width / 1000;
-      // According to https://en.wikipedia.org/wiki/Em_(typography)
-      // an average cap height should be 70% of 1em
-      const capHeight = 0.7 * em;
-      // 1.5 * capHeight * fontSize seems to be a good value for lineHeight
-      fontSize = Math.max(1, Math.floor(height / (1.5 * capHeight)));
+      if (lineCount === -1) {
+        fontSize = Math.round(0.8 * height * 10) / 10;
```

Comment:
The value 0.8 used quite some times in this function should become a constant with a descriptive name of what it does. Just looking at the code doesn't allow me to figure out what its purpose is and why it's 0.8. In general this function is a bit "magic" due to the magic numbers but also because of for example this calculation, so maybe a comment would be good here to explain the logic in the various branches.

Reply:
Yeah agreed with the "magic" aspect. It just looks good with this value (at least for my eyes).
The current way to produce obviously too big font and I guess the 1.5 I found is line height in a paragraph and not for an input.